### DISCUSSION TITLE
image preprocessing

### DISCUSSION THREAD

**Discussion by omkarsingh1008** [mindee/doctr#718] [Q&A]
which type of image preprocessing you did in Doctr?

**Comment by fg-mindee** [comment]
Hi @omkarsingh1008 :wave: 

Better to show you:
```python
import os 

os.environ['USE_TORCH'] = '1'

from doctr.models import detection_predictor, recognition_predictor

print(detection_predictor().pre_processor)
print(recognition_predictor().pre_processor)
```

yields:
```
PreProcessor(
  (resize): Resize(output_size=(1024, 1024), interpolation='bilinear')
  (normalize): Normalize(mean=(0.798, 0.785, 0.772), std=(0.264, 0.2749, 0.287))
)
PreProcessor(
  (resize): Resize(output_size=(32, 128), interpolation='bilinear', preserve_aspect_ratio=True, symmetric_pad=False)
  (normalize): Normalize(mean=(0.694, 0.695, 0.693), std=(0.299, 0.296, 0.301))
)
```

I hope this answers your question :)

**Reply by omkarsingh1008** [reply]
not this type of preprocessing did you use any image enhancement like dilation, otsu, etc

**Reply by fg-mindee** [reply]
? If you indeed mean preprocessing for inference, my previous answer still stands :upside_down_face: Those are the only preprocessing steps that are taken for inference.

If you mean data augmentation, you can check all our training scripts over here: https://github.com/mindee/doctr/tree/main/references (all augmentations are defined in the script)

**Reply by omkarsingh1008** [reply]
like when we working on scan documents that time we have to remove some noise from the image. did you use in doctr any noise removal technique

**Reply by fg-mindee** [reply]
> Those are the only preprocessing steps that are taken for inference.

As of now, no :point_up:
But we might include this in the near future :ok_hand:
